plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/server/FileBasedWorkspaceInitializer.java

```diff
@@ -75,16 +82,97 @@ public IWorkspaceConfig getWorkspaceConfig(URI workspaceBaseURI) {
 				allProjectURIs.add(new FileURI(path.toFile()));
 			}
 
-			headlessHelper.registerProjectsToFileBasedWorkspace(allProjectURIs, workspace);
+			registerProjectsToFileBasedWorkspace(allProjectURIs);
 
 			return new N4JSWorkspaceConfig(n4jsCore);
 
-		} catch (N4JSCompileException e) {
-			e.printStackTrace();
-			throw new RuntimeException(e);
 		} finally {
 			this.knownWorkspaceBaseURI = workspaceBaseURI;
 		}
 	}
 
+	/**
+	 * Registers provided project URIs in the given workspace.
+	 *
+	 * Skips {@link IN4JSProject}s that are already registered with the given {@code workspace}.
+	 *
+	 */
+	public void registerProjectsToFileBasedWorkspace(Iterable<FileURI> projectURIs) {
```

Comment:
This is only called from `#getWorkspaceConfig()` above, never from the outside. Would it make sense to make this private?